DevOps.Util/BaseServer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DevOps.Util
+{
+    public class BaseServer
```

Comment:
Why did you want to add a base server here? Since you started this PR I essentially gave up on the idea of having a layer below `DevOpsServer`. It ended up always being the wrong abstraction because I was trying to insert items like caching below the server. Ended up being a lot better to introduce them above `DevOpsServer` and just let that be a simple interface into the AzDO servers.

In the cases I needed to hook it for testing I just hook the underlying `HttpClient` itself now.